ringpop.go

```diff
@@ -213,6 +233,20 @@ func (rp *Ringpop) RegisterListener(l EventListener) {
 	rp.listeners = append(rp.listeners, l)
 }
 
+// getState gets the state of the current Ringpop instance.
+func (rp *Ringpop) getState() state {
+	rp.stateMutex.Lock()
+	defer rp.stateMutex.Unlock()
```

Comment:
Nice, good tip to know. I'll change it. Is there a good source/reference showing the performance impact of defer?